lib/util/text_parser.js

```diff
@@ -20,7 +20,7 @@ shaka.util.TextParser = class {
      * @const
      * @private {string}
      */
-    this.data_ = data;
+    this.data_ = data || '';
```

Comment:
The argument is a non-nullable type, so `data` will never be null.  If this is needed, you probably are getting compiler errors elsewhere where you are passing in null.  Be sure to run `./build/all.py`.

Reply:
You'd think so, but if you look at the [demo](https://shaka-player-demo.appspot.com/demo/#audiolang=en-US;textlang=en-US;uilang=en-US;asset=https://bloomtest.nyc3.digitaloceanspaces.com/roles/test5.mpd;panel=CUSTOM%20CONTENT;build=uncompiled) (use Italian captions) I made to showcase this problem, you'll see an undefined value makes it into `data_`. I've verified myself that the compiler doesn't catch anything with `all.py`.
What happens is `parseCue_` thinks `text[0]` is a cue ID because it has no `-->`, so it splices that element out. If the cue only had one element, handing the new `text[0]` to `TextParser` gives it an undefined value. Coalescing values handed to the `TextParser` constructor lets you see the 2001 exception that was intended to be thrown from `VttTextParser`.

I assumed that this is just a limitation of the Closure type system i.e. it doesn't account for every peculiarity of JavaScript.

Reply:
What we're saying is that the compiler does not allow "undefined" here.  If you are able to get "undefined" into the "data" argument, then either the compiler has failed to catch your error or you haven't run the compiler.  Please run ./build/all.py, because we would expect the compiler to point out any calls to this method that would pass "undefined" for "data".

Reply:
I use all.py before every push, and this issue is not generated from any modification of mine. The bug is present in the v2.5.9 release, which can be seen at [shaka-player-demo.appspot.com](https://shaka-player-demo.appspot.com/demo/#audiolang=en-US;textlang=en-US;uilang=en-US;asset=https://bloomtest.nyc3.digitaloceanspaces.com/roles/test5.mpd;panel=CUSTOM%20CONTENT;build=uncompiled). I assume you trust the official demo has gone through the full build process.

![image](https://user-images.githubusercontent.com/4093170/74556177-75659100-4f2b-11ea-8f9d-be22c42c09fb.png)

For the compiler to guarantee you can't hand undefined values to a method, it must either prove the element exists using the information it has at compile time (impossible here since the data is dynamic), or it must force the check on you in some way. The Closure compiler seems to not be that strict. Somewhat comically, even a simple thing like this builds:
```js
    const undef = [''];
    undef.pop();
    const parser = new shaka.util.TextParser(undef[1]);
```
If you remove the pop then this will be caught as `Array index out of bounds`, but with it in place the static analyzer seems to give up on verifying the type is correct.